connectors/opa/main.go

```diff
@@ -1,71 +1,101 @@
 // Copyright 2021 IBM Corp.
 // SPDX-License-Identifier: Apache-2.0
 
-/*
- * Policy Manager Service
- *
- * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
- *
- * API version: 1.0.0
- * Generated by: OpenAPI Generator (https://openapi-generator.tech)
- */
-
 package main
 
 import (
-	"log"
+	"fmt"
 	"os"
-	"strconv"
+	"strings"
 	"time"
 
-	openapiserver "fybrik.io/fybrik/connectors/opa/go"
-	opabl "fybrik.io/fybrik/connectors/opa/lib"
+	"emperror.dev/errors"
 	"fybrik.io/fybrik/pkg/connectors/datacatalog/clients"
-	"github.com/hashicorp/go-retryablehttp"
+	"fybrik.io/fybrik/pkg/environment"
+	"github.com/gin-gonic/gin"
+	"github.com/spf13/cobra"
+)
+
+const (
+	envOPAServerURL        = "OPA_SERVER_URL"
+	envCatalogConnectorURL = "CATALOG_CONNECTOR_URL"
+	envCatalogProviderName = "CATALOG_PROVIDER_NAME"
+	envConnectionTimeout   = "CONNECTION_TIMEOUT"
 )
 
-func getEnv(key string) string {
-	value, exists := os.LookupEnv(key)
-	if !exists {
-		log.Fatalf("Env Variable %v not defined", key)
+// NewRouter returns a new router.
+func NewRouter(controller *ConnectorController) *gin.Engine {
+	router := gin.Default()
+	router.POST("/getPoliciesDecisions", controller.GetPoliciesDecisions)
+	return router
+}
+
+// RootCmd defines the root cli command
+func RootCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "opa",
+		Short: "Kubernetes based policy manager connector for Fybrik",
 	}
-	log.Printf("Env. variable extracted: %s - %s\n", key, value)
-	return value
+	cmd.AddCommand(RunCmd())
+	return cmd
 }
 
-func main() {
-	log.Println("in main of V2 OPA Connector: ")
-	catalogConnectorAddress := getEnv("CATALOG_CONNECTOR_URL")
-	catalogProviderName := getEnv("CATALOG_PROVIDER_NAME")
+// RunCmd defines the command for running the connector
+func RunCmd() *cobra.Command {
+	ip := ""
+	port := 8080
+	cmd := &cobra.Command{
+		Use:   "run",
```

Comment:
Need to add to dockerfile `CMD [ "run" ]`

Reply:
added